sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -169,11 +192,56 @@ template <> inline void print<>(std::nullptr_t val) { print<void *>(val); }
 inline void printArgs(void) {}
 template <typename Arg0, typename... Args>
 void printArgs(Arg0 arg0, Args... args) {
-  std::cout << "       ";
+  std::cout << "\t";
   print(arg0);
   pi::printArgs(std::forward<Args>(args)...);
 }
 
+template <typename T> struct printOut {
+  printOut(T val) {}
+}; // Do nothing
+
+template <> struct printOut<PiEvent *> {
+  printOut(PiEvent *val) {
+    std::cout << "\t[out]pi_event * : " << val;
```

Comment:
Why do we need printOut and print<event> with exactly the same functionality?

Reply:
This one is adding "[out]" in the print of values, but more importantly the PrintOut is only acting on pointer types